### ISSUE TITLE
I/O timeouts

### ISSUE THREAD

**Issue by eugeneia** [usocket/usocket#8]
I am wondering if it would be possible for usocket to support I/O timeouts on implementations that support it? This would really add a lot of value to me. I would volunteer to work on supporting Clozure CL.

**Comment by binghe** [comment]
I think we do, through API function SOCKET-OPTION (option.lisp)

**Comment by eugeneia** [comment]
Oh I wasn't aware of `socket-option`. Would you be interested in me helping to update the usocket documentation?

**Comment by binghe** [comment]
Sure, if that's possible. Thanks.  Also, currently I only provided "receive timeout" as socket options, do you think it's also necessary to provide "send timeout?

**Comment by eugeneia** [comment]
Yes, I think send timeout (`SO_SNDTIMEO`) is important to have, as well as connection timeout (???) and keep-alive (`SO_KEEPALIVE`).

**Comment by hanshuebner** [comment]
Connect timeouts need to be implemented explicitly by using asynchronous
I/O calls or timers/signals.
Am 20.08.2015 1:30 nachm. schrieb "Max Rottenkolber" <
notifications@github.com>:

**Comment by eugeneia** [comment]
@hanshuebner Clozure CL at least has the option in `make-socket`. Could be that other implementation support it as well so that it could be a keyword parameter to `usocket:socket-connect`.

**Comment by tmccombs** [comment]
Sbcl has a timeout option for socket-make-stream. Also, for sbcl at least, the receive-timeout option only works for stream sockets, not datagram sockets.

**Comment by fjl** [comment]
Adding some tests in #77 to see if the timeouts provided by usocket actually work.
